### PR TITLE
#3914-app - Avoid numeric values with too many trailing zeros

### PR DESCRIPTION
**Pull request by metas-ts**
Strip trailing zeros if a given BigDecimal value has a scale of >= 15
#3914

### CODE REVIEW

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java [hunk 1]
```diff
@@ -1055,55 +1059,65 @@ else if (oldValue != null)
...
+			{
+				m_newValues[index] = valueToUse;     // correct
+			}
 			// Integer can be set as BigDecimal
-			else if (value.getClass() == BigDecimal.class
+			else if (valueToUse.getClass() == BigDecimal.class
 					&& p_info.getColumnClass(index) == Integer.class)
-				m_newValues[index] = new Integer(((BigDecimal)value).intValue());
+			{
+				m_newValues[index] = new Integer(((BigDecimal)valueToUse).intValue());
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java]: don't use `new Integer`

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java [hunk 2]
```diff
@@ -1055,55 +1059,65 @@ else if (oldValue != null)
...
+			}
 			// Set Boolean
 			else if (p_info.getColumnClass(index) == Boolean.class
-					&& ("Y".equals(value) || "N".equals(value)))
-				m_newValues[index] = Boolean.valueOf("Y".equals(value));
+					&& ("Y".equals(valueToUse) || "N".equals(valueToUse)))
+			{
+				m_newValues[index] = Boolean.valueOf("Y".equals(valueToUse));
```
Comment (teosarca) [inline de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java]: use `DisplayType.toBoolean`

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java [hunk 3]
```diff
@@ -1233,6 +1261,24 @@ else if (p_info.getColumnClass(index) == Integer.class
 		return true;
 	}   // set_ValueNoCheck
 
+	private Object stripZerosIfBigDecimalScaleTooBig(@Nullable final Object value)
```
[the remark this answers is not in the document]
Comment (metas-ts) [reply de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java]: then i can't log `this` anymore..or would you prefer it being static and this being passed as paramter?

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java [hunk 4]
```diff
@@ -1157,64 +1173,76 @@ public final boolean set_ValueNoCheck(final String ColumnName, final Object valu
...
+			{
+				m_newValues[index] = valueToUse;     // correct
+			}
 			// Integer can be set as BigDecimal
-			else if (value.getClass() == BigDecimal.class
+			else if (valueToUse.getClass() == BigDecimal.class
 					&& p_info.getColumnClass(index) == Integer.class)
-				m_newValues[index] = new Integer(((BigDecimal)value).intValue());
+			{
+				m_newValues[index] = new Integer(((BigDecimal)valueToUse).intValue());
```
[the remark this answers is not in the document]
Comment (metas-ts) [reply de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java]: done, also at other places

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java (hunk 2 above)
Comment (metas-ts) [reply de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java]: i'll rather use StringUtils.toBoolean here..

Comment (teosarca) [reply de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java]: yup

File: de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (teosarca) [reply de.metas.adempiere.adempiere/base/src/main/java-legacy/org/compiere/model/PO.java]: you are right. let it like it is.

**Pull request merged**